src/WingetCreateCLI/PromptHelper.cs

```diff
@@ -0,0 +1,270 @@
+                    var value = Prompt.List<int>(message, minimum: minimum);
+                    property.SetValue(model, value);
+                }
+                else if (elementType.IsEnum)
+                {
+                    var value = Prompt.MultiSelect(message, Enum.GetNames(elementType), minimum: 0);
+
+                    if (value.Any())
+                    {
+                        Type genericListType = typeof(List<>).MakeGenericType(elementType);
+                        var enumList = (IList)Activator.CreateInstance(genericListType);
+
+                        foreach (var item in value)
+                        {
+                            var enumItem = Enum.Parse(elementType, item);
+                            enumList.Add(enumItem);
+                        }
+
+                        property.SetValue(model, enumList);
+                    }
+                }
+                else if (elementType == typeof(PackageDependencies))
+                {
+                    List<PackageDependencies> packageDependencies = (List<PackageDependencies>)property.GetValue(model) ?? new List<PackageDependencies>();
+
+                    PromptForPackageDependencies(packageDependencies);
+                    if (packageDependencies.Any())
+                    {
+                        property.SetValue(model, packageDependencies);
+                    }
+                    else
+                    {
+                        property.SetValue(model, null);
+                    }
+                }
+            }
+            else
+            {
+                if (property.PropertyType == typeof(InstallerSwitches))
+                {
+                    InstallerSwitches installerSwitches = (InstallerSwitches)property.GetValue(model) ?? new InstallerSwitches();
+                    PromptSubfieldProperties(installerSwitches, property, model);
+                }
+                else if (property.PropertyType == typeof(Dependencies))
+                {
+                    Dependencies dependencies = (Dependencies)property.GetValue(model) ?? new Dependencies();
+                    PromptSubfieldProperties(dependencies, property, model);
+                }
+            }
+        }
+
+        private static void PromptEnum(object model, PropertyInfo property, Type enumType, bool required, string message, string defaultValue = null)
+        {
+            var enumList = Enum.GetNames(enumType);
+
+            if (!required)
+            {
+                enumList.Append(Resources.None_MenuItem);
+            }
+
+            var value = Prompt.Select(message, enumList, defaultValue: defaultValue);
+            if (value != Resources.None_MenuItem)
+            {
+                property.SetValue(model, Enum.Parse(enumType, value));
+            }
+        }
+
+        private static void PromptForPackageDependencies(List<PackageDependencies> packageDependencies)
```

Comment:
>PromptForPackageDependencies [](http://example.com/codeflow?start=28&length=28)

Can this be made more generic? I'm thinking a method that takes in a List<T> and a resource message where T : null()?

Not totally sure that'd work, but it'd be cool to have this method already be flexible to use with other future types

Reply:
This was a lot harder to do, but I think I got it working, all the prompting methods should be generic now.

Reply:
Wow, above and beyond! :)